frontend/apps/pablo/defi/utils/pablo/pools/stats.ts

```diff
@@ -1,85 +1,154 @@
+      (acc, cur) => {
+        acc.liquidity = acc.liquidity.plus(
+          fromChainUnits(cur.totalLiquidity.toString())
+        );
+        acc.volume = acc.volume.plus(
+          fromChainUnits(cur.totalVolume.toString())
+        );
 
-        pabloPools = pabloPools.map((poolState: any) => {
-            return {
-                totalLiquidity: fromChainUnits(poolState.totalLiquidity),
-                totalVolume: fromChainUnits(poolState.totalVolume),
-                totalFees: fromChainUnits(poolState.totalFees),
-                calculatedTimestamp: Number(poolState.calculatedTimestamp),
-                transactionCount: Number(poolState.transactionCount),
-                quoteAssetId: Number(poolState.quoteAssetId),
-                poolId: Number(poolState.poolId),
-            };
-        });
+        return acc;
+      },
+      { liquidity: new BigNumber(0), volume: new BigNumber(0) }
+    );
+    stats ||= {};
+    stats[assetId] = {
+      total,
+      asset,
+      spotPrice: new BigNumber(0),
+    };
 
-        return pabloPools;
-    } catch (err: any) {
-        console.error(err.message);
-        return [];
+    if (otherAsset) {
+      const assetPrice = getOraclePrice(asset.getSymbol(), "coingecko", "usd");
+      const otherAssetPrice = getOraclePrice(
+        otherAsset.getSymbol(),
+        "coingecko",
+        "usd"
+      );
+      // If we have other asset, we can calculate the price based on the pair and ratio
+      if (assetPrice.isZero() && !otherAssetPrice.isZero()) {
+        const spotPriceResponse = await querySpotPriceFromPool(
+          pool.poolId.toString(),
+          asset.getPicassoAssetId().toString(),
+          otherAsset.getPicassoAssetId().toString()
+        );
+        const spotPrice =
+          spotPriceResponse.data?.pabloSpotPrice?.spotPrice || 0;
+        stats[assetId] = {
+          total,
+          asset,
+          spotPrice: new BigNumber(spotPrice).multipliedBy(otherAssetPrice),
+        };
+      }
+      // If we don't have the other asset, and current assetPrice is zero, resolve to 0
+      else {
+        stats[assetId] = {
+          total,
+          asset,
+          spotPrice: asset.getPrice(),
+        };
+      }
     }
-};
-
+  }
+  return stats;
+}
 
-export function calculatePoolStats(data: PabloPoolQueryResponse[]):
-    | {
-        _24HrFee: string;
-        _24HrVolume: string;
-        totalVolume: string;
-        _24HrTransactionCount: number;
-        poolId: number;
-    }
-    | undefined {
-    if (!data.length) return undefined;
+export function getPoolTVL(stats: GetStatsReturn): BigNumber {
+  if (stats === null) {
+    return new BigNumber(0);
+  }
 
-    let yesterday = data[0].calculatedTimestamp - 1 * DAYS;
-    const yesterdayState = data.find((i) => i.calculatedTimestamp < yesterday);
+  return Object.values(stats).reduce((acc, cur) => {
+    acc = acc.plus(cur.total.volume.multipliedBy(cur.spotPrice));
```

Comment:
This seems off, why we are taking volume for TVL?

Reply:
ah it should be the other way around right?